resources/testharness.js

```diff
@@ -1509,6 +1509,23 @@ policies and contribution forms [3].
     }
     expose(assert_any, "assert_any");
 
+    function assert_precondition(precondition, description) {
+        if (!precondition) {
+            throw new PreconditionFailedError(description);
+        }
+    }
+    expose(assert_precondition, "assert_precondition");
+
+
+    function PreconditionFailedError(message)
+    {
+        AssertionError.call(this, message);
+    }
+    expose(PreconditionFailedError, "PreconditionFailedError");
+
+    PreconditionFailedError.prototype = Object.create(AssertionError.prototype);
+    PreconditionFailedError.prototype.constructor = PreconditionFailedError;
```

Comment:
For some reason, `new PreconditionFailedError('foo') instanceof AssertionError` is false.

Reply:
The style of inheritence used here looks unfamiliar to me, is it used elsewhere in testharness.js and is `instanceof` correct in those cases?

Reply:
It's es5 "inheritance" and yes, `AssertionError` uses it for inheriting `Error`

Reply:
It's overriding `Foo.prototype.constructor` that looks unfamiliar. Maybe this pattern just isn't right, or is `new AssertionError('foo') instanceof Error` true?

Reply:
So, the issue was that the whole thing is in a closure. It uses `expose` to add the definitions to the global_scope. As a result, the references need to exist before calling expose, otherwise they'll refer to a local of the same name, and the globally resolved `AssertionError` is a different reference, thus instanceof is false. Fixed by declaring the types in the right order.